tests/test-desktop-files.lua

```diff
@@ -0,0 +1,17 @@
+local runner = require("_runner")
+local menubar = require("menubar")
+local gears_debug = require("gears.debug")
+
+local assets_dir = os.getenv("TEST_ASSETS") .. "/"
+menubar.menu_gen.all_menu_dirs = { assets_dir }
+
+runner.run_steps({
+    function()
+        print("Running desktop file test!")
+        menubar.menu_gen.generate(function(menu_entries)
+            print("Menu list generated!")
+            gears_debug.dump(menu_entries, "menu_entries", 4)
+        end)
```

Comment:
The test are executed up to 5 time if they return nil. In case you have something async, just return nil and it will be executed again 100ms later. So, execute the code, return true without further check and in the next test step, check the code. Do not assert, just return nil. If the test don't return true after the last attempt, it will be considered a failure anyway.